trino/client.py

```diff
@@ -127,6 +138,9 @@ def __init__(
         self._roles = roles.copy() if roles is not None else {}
         self._prepared_statements: Dict[str, str] = {}
         self._object_lock = threading.Lock()
+        if timezone:  # Check timezone validity
+            ZoneInfo(timezone)
```

Comment:
Just for other reviewers this throws on invalid timezones (invalidity is determined by the OS's tzdb files or the tzdata package shipped by Python itself):

```
ZoneInfoNotFoundError: 'No time zone found with key Foobar'
```